keras/engine/training_test.py

```diff
@@ -2929,6 +2930,26 @@ def test_metric_state_reset_between_fit_and_evaluate(self):
     model.evaluate(x_test, y_test, batch_size=5)
     self.assertEqual(self.evaluate(acc_obj.count), 10)
 
+  @keras_parameterized.run_all_keras_modes
+  def test_metric_state_reset_between_test_on_batch_and_evaluate(self):
+    model = sequential.Sequential()
+    model.add(layers_module.Dense(3, activation='relu', input_dim=4))
+    model.add(layers_module.Dense(1, activation='sigmoid'))
+    acc_obj = metrics_module.BinaryAccuracy()
+    model.compile(
+        loss='mae',
+        metrics=[acc_obj],
+        optimizer=RMSPropOptimizer(learning_rate=0.001),
+        run_eagerly=testing_utils.should_run_eagerly())
+
+    x_test = np.random.random((10, 4))
+    y_test = np.random.random((10, 1))
+    loss, acc = model.test_on_batch(x_test[:2],y_test[:2])
+    model.evaluate(x_test, y_test)
+    loss_1, acc_1 = model.test_on_batch(x_test[:2],y_test[:2])
```

Comment:
Could u add another model.eval below this line, and make sure the output are the same as the first model.eval()?

This will make sure we don't include the any result from model.test_on_batch().

Reply:
Done